### PR TITLE
Reader Refresh: Choose random tag stream header image

### PR DESCRIPTION
**Pull request by bluefuton**
Currently, the same tag header is showed every time the same tag page is visited. 

This PR greets the user with random image from a collection of 10 returned by the tag image API.

screen shot 2016-12-02 at 13 02 02

### CODE REVIEW

**Comment by bluefuton** [comment]
Todo: 
- [ ] make image choice somewhat sticky (don't keep grabbing random images on every render)

File: client/reader/tag-stream/header.jsx [hunk 1]
```diff
@@ -5,77 +5,106 @@ import React from 'react';
...
-						onFollowToggle={ onFollowToggle } />
-				</div>
-			}
+	componentWillReceiveProps( nextProps ) {
+		this.pickNewTagImage( nextProps );
+	}
+
+	pickNewTagImage = ( props = this.props ) => {
+		if ( this.state && this.state.tagImages !== props.tagImages ) {
+			this.state = {
```
Comment (blowery) [inline client/reader/tag-stream/header.jsx]: this _shouldn't_ work, the only place you can set `this.state` safely is in the constructor. I'd rejigger this to have pickNewImage return the chosen one and have cwrp call it and then call setState if the chosen one changed. pickNewTagImage would also have to return `null` if no `tagImages` are passed in...

File: client/reader/tag-stream/header.jsx [hunk 2]
```diff
@@ -5,77 +5,106 @@ import React from 'react';
...
+			'is-placeholder': isPlaceholder,
+			'has-back-button': hasBackButton
+		} );
+		const imageStyle = {};
+		const tagImage = this.state.chosenTagImage;
+
+		if ( tagImage ) {
+			const imageUrl = resizeImageUrl( 'https://' + tagImage.url, { resize: `${ TAG_HEADER_WIDTH },${ TAG_HEADER_HEIGHT }` } );
+			const safeCssUrl = cssSafeUrl( imageUrl );
+			imageStyle.backgroundImage = 'url(' + safeCssUrl + ')';
```
Comment (blowery) [inline client/reader/tag-stream/header.jsx]: maybe we should make up a `<BackgroundImage/>` component that does all this for us. We seem to keep doing it.

Comment (blowery) [inline client/reader/tag-stream/header.jsx]: leave this for another PR

**Comment by samouri** [comment]
With the latest version its being jumpy 🐰 :
movingabout

**Comment by blowery** [comment]
? I can't repro that... How are you switching tags?

**Comment by blowery** [comment]
@samouri can you try repulling and a restart of calypso?

**Comment by samouri** [comment]
I was using calypso.live.   I was clicking the tabs on the left sidebar.
I must have been using a version from before your commit though because I can't repro it now either  👍.  Don't mind me!
